mungegithub/mungers/inactive-review-handler.go

```diff
@@ -0,0 +1,169 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+	"k8s.io/test-infra/mungegithub/features"
+	"k8s.io/test-infra/mungegithub/github"
+	"time"
+)
+
+const (
+	msgHeader   = "Sorry it is taking a long time for @%s to review your PR. This person may be on vacation or otherwise occupied. "
+	msgExpedite = "To expedite a review, please "
+)
+
+type InactiveReviewHandler struct {
+	features *features.Features
+}
+
+func init() {
+	h := &InactiveReviewHandler{}
+	RegisterMungerOrDie(h)
+}
+
+// Name is the name usable in --pr-mungers
+func (i *InactiveReviewHandler) Name() string { return "inactive-review-handler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (i *InactiveReviewHandler) RequiredFeatures() []string {
+	return []string{features.RepoFeatureName, features.AliasesFeature}
+}
+
+// Initialize will initialize the munger
+func (i *InactiveReviewHandler) Initialize(config *github.Config, features *features.Features) error {
+	i.features = features
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (i *InactiveReviewHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (i *InactiveReviewHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// Determine whether the PR is active based on number of seconds
+// A PR is inactive if it is not being modified for more than one week
+// A week has 604800 seconds
+func (i *InactiveReviewHandler) isPRActive(prCreatedAt *time.Time, comments []*githubapi.PullRequestComment) bool {
+	lastUpdateTime, ok := findLastHumanPullRequestUpdateHelper(prCreatedAt, comments)
+
+	//treat the PR as active for early return if failed to obtain last update time
+	if !ok {
+		return true
+	}
+
+	if time.Now().Sub(*lastUpdateTime).Seconds() <= 604800 {
+		return true
+	}
+
+	return false
+}
+
+// Suggest a new reviewer who is NOT any of the existing reviewers
+// (1) get all current assignees for the PR
+// (2) get potential owners of the PR using Blunderbuss algorithm (calling getPotentialOwners() function)
+// (3) filter out current assignees from the potential owners
+// (4) if there is no any new reviewer available, the bot will encourage the PR author to ping all existing assignees
+// (5) otherwise, select a new reviewer using Blunderbuss algorithm (calling selectMultipleOwners() function with number of assignees parameter of one)
+// Note: the munger will suggest a new reviewer when the PR currently does not have any reviewer
+func (i *InactiveReviewHandler) suggestNewReviewer(issue *githubapi.Issue, potentialOwners weightMap, weightSum int64) string {
+	var newReviewer string
+
+	if len(potentialOwners) > 0 && issue.Assignees != nil && len(issue.Assignees) > 0 {
+		oldReviewers := issue.Assignees
```

Comment:
I mean on line 94 you can use

`for _, oldReviewer := range issue.Assignees {`